windows/devices/uwb/include/windows/devices/uwb/UwbDeviceConnector.hxx

```diff
@@ -192,8 +195,23 @@ private:
     void
     OnSessionRangingData(::uwb::protocol::fira::UwbRangingData rangingData);
 
+    /**
+     * @brief Internal function to check if there are callbacks present
+     *
+     * @return true
+     * @return false
+     */
+    bool
+    CallbacksPresent();
+
 private:
+    mutable std::shared_mutex m_eventCallbacksGate;

```

Comment:
Why is the `mutable` keyword needed?

Reply:
This is typically done such that non-const locking functions can be used in `const` functions. It's technically cheating since the lock state does change, but the idea of `const` is more meant to reflect actual object state and not pseudo-state used to control concurrency.